src/accordion.tsx

```diff
@@ -0,0 +1,307 @@
+        return value === undefined ? [] : typeof value === 'number' ? [value] : value;
+    };
+
+    const [index, setIndex] = React.useState<Array<number>>(getValueAsList(defaultValue));
+
+    React.useEffect(() => setIndex(getValueAsList(defaultValue)), [defaultValue]);
+
+    React.useEffect(() => {
+        if (index.length > 1 && singleOpen) {
+            index.splice(1);
+            setIndex([...index]);
+        }
+    }, [singleOpen, index]);
+
+    const updateIndexOnToogle = (item: number, index?: Array<number>) => {
+        if (!index) {
+            return [item];
+        }
+
+        const valueIndex = index.indexOf(item);
+        if (valueIndex === -1) {
+            if (singleOpen) {
+                index = [item];
+            } else {
+                index.push(item);
+            }
+        } else {
+            index.splice(valueIndex, 1);
+        }
+
+        return [...index];
+    };
+
+    const toggle = (item: number) => {
+        if (!isControlledByParent) {
+            setIndex(updateIndexOnToogle(item, index));
+        }
+        if (onChange) {
+            const currentItemValue = (isControlledByParent ? getValueAsList(value) : index).includes(item);
+            onChange(item, !currentItemValue);
+        }
+    };
+
+    if (isControlledByParent) {
+        return [getValueAsList(value), toggle];
+    }
+
+    return [index, toggle];
+};
+
+const getAccordionItemIndex = (element: Element | null) => {
+    const accordionAncestor = element?.closest('[data-accordion]');
+    if (!accordionAncestor) return undefined;
+
+    return Array.from(accordionAncestor.querySelectorAll('[data-accordion-item]'))
+        .filter((e) => e.closest('[data-accordion]') === accordionAncestor)
+        .findIndex((e) => e === element);
+};
+
+const AccordionItemContent = React.forwardRef<TouchableElement, AccordionItemContentProps>(
+    ({content, dataAttributes, trackingEvent, ...props}, ref) => {
+        const panelContainerRef = React.useRef<HTMLDivElement | null>(null);
+        const itemRef = React.useRef<HTMLDivElement | null>(null);
+        const {index, toogle} = useAccordionContext();
+        const isInverse = useIsInverseVariant();
+        const labelId = useAriaId();
+        const panelId = useAriaId();
+
+        const [itemIndex, setItemIndex] = React.useState<number>();
+        const isOpen = itemIndex !== undefined && index?.includes(itemIndex);
+
+        React.useEffect(() => {
+            setItemIndex(getAccordionItemIndex(itemRef.current));
+        }, []);
+
+        return (
+            <div ref={itemRef} {...getPrefixedDataAttributes({...dataAttributes, 'accordion-item': true})}>
+                <BaseTouchable
+                    ref={ref}
+                    className={classNames(styles.itemContent, {
+                        [styles.touchableBackground]: !isInverse,
+                        [styles.touchableBackgroundInverse]: isInverse,
```

Comment:
this could be a ternary, because both classes are exclusive

Reply:
updated